src/Rules/Deprecations/InheritanceOfDeprecatedClassRule.php

```diff
@@ -0,0 +1,77 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Deprecations;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Class_;
+use PHPStan\Analyser\Scope;
+use PHPStan\Broker\Broker;
+
+class InheritanceOfDeprecatedClassRule implements \PHPStan\Rules\Rule
+{
+
+	/**
+	 * @var Broker
+	 */
+	private $broker;
+
+	public function __construct(Broker $broker)
+	{
+		$this->broker = $broker;
+	}
+
+	public function getNodeType(): string
+	{
+		return Class_::class;
+	}
+
+	/**
+	 * @param Class_ $node
+	 * @param \PHPStan\Analyser\Scope $scope
+	 * @return string[] errors
+	 */
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if ($node->extends === null) {
+			return [];
+		}
+
+		$errors = [];
+
+		$className = isset($node->namespacedName)
+			? (string) $node->namespacedName
+			: (string) $node->name;
+
+		try {
+			$class = $this->broker->getClass($className);
+		} catch (\PHPStan\Broker\ClassNotFoundException $e) {
+			return [];
+		}
+
+		$parentClassName = (string) $node->extends;
+
+		try {
+			$parentClass = $this->broker->getClass($parentClassName);
+
+			if ($parentClass->isDeprecated()) {
+				if (!$class->getNativeReflection()->isAnonymous()) {
+					$errors[] = sprintf(
+						'Inheritance of deprecated class %s in class %s.',
```

Comment:
What about `Class %s extends deprecated class %s.`?

Reply:
Also, what about interfaces extending other deprecated interfaces? 😊

Reply:
> Also, what about interfaces extending other deprecated interfaces? 😊

That should be handled in the `ImplementationOfDeprecatedInterfaceRule`. But tests are missing! Haven't thought about that.

Reply:
I don't think so, you need a rule that processes `Interface_::class` for that.

Reply:
You're right 🙂
I'll make sure it works!